vm.go

```diff
@@ -0,0 +1,131 @@
+	ExtfragThreshold          int64 `fileName:"extfrag_threshold"`
+	HugetlbShmGroup           int64 `fileName:"hugetlb_shm_group"`
+	LaptopMode                int64 `fileName:"laptop_mode"`
+	LegacyVaLayout            int64 `fileName:"legacy_va_layout"`
+	// LowmemReserveRatio        [?]int64 `fileName:"lowmem_reserve_ratio"`
+	MaxMapCount            int64  `fileName:"max_map_count"`
+	MemoryFailureEarlyKill int64  `fileName:"memory_failure_early_kill"`
+	MemoryFailureRecovery  int64  `fileName:"memory_failure_recovery"`
+	MinFreeKbytes          int64  `fileName:"min_free_kbytes"`
+	MinSlabRatio           int64  `fileName:"min_slab_ratio"`
+	MinUnmappedRatio       int64  `fileName:"min_unmapped_ratio"`
+	MmapMinAddr            int64  `fileName:"mmap_min_addr"`
+	NrHugepages            int64  `fileName:"nr_hugepages"`
+	NrHugepagesMempolicy   int64  `fileName:"nr_hugepages_mempolicy"`
+	NrOvercommitHugepages  int64  `fileName:"nr_overcommit_hugepages"`
+	NumaStat               int64  `fileName:"numa_stat"`
+	NumaZonelistOrder      string `fileName:"numa_zonelist_order"`
+	OomDumpTasks           int64  `fileName:"oom_dump_tasks"`
+	OomKillAllocatingTask  int64  `fileName:"oom_kill_allocating_task"`
+	OvercommitKbytes       int64  `fileName:"overcommit_kbytes"`
+	OvercommitMemory       int64  `fileName:"overcommit_memory"`
+	OvercommitRatio        int64  `fileName:"overcommit_ratio"`
+	PageCluster            int64  `fileName:"page-cluster"`
+	PanicOnOom             int64  `fileName:"panic_on_oom"`
+	PercpuPagelistFraction int64  `fileName:"percpu_pagelist_fraction"`
+	StatInterval           int64  `fileName:"stat_interval"`
+	Swappiness             int64  `fileName:"swappiness"`
+	UserReserveKbytes      int64  `fileName:"user_reserve_kbytes"`
+	VfsCachePressure       int64  `fileName:"vfs_cache_pressure"`
+	WatermarkBoostFactor   int64  `fileName:"watermark_boost_factor"`
+	WatermarkScaleFactor   int64  `fileName:"watermark_scale_factor"`
+	ZoneReclaimMode        int64  `fileName:"zone_reclaim_mode"`
+}
+
+// Vm reads the Vm statistics from the specified `proc` filesystem.
+func (fs FS) Vm() (Vm, error) {
+	path := fs.proc.Path("sys/vm")
+	file, err := os.Stat(path)
+	if err != nil {
+		return Vm{}, err
+	}
+	if !file.Mode().IsDir() {
+		return Vm{}, fmt.Errorf("%s is not a directory", path)
+	}
+
+	return parseVm(path)
+}
+
+// parseVm scans predefined files in /proc/sys/vm
+// directory and gets their contents.
+func parseVm(devicePath string) (Vm, error) {
+	interfaceClass := Vm{}
+	interfaceElem := reflect.ValueOf(&interfaceClass).Elem()
+	interfaceType := reflect.TypeOf(interfaceClass)
+
+	for i := 0; i < interfaceElem.NumField(); i++ {
+		fieldType := interfaceType.Field(i)
+		fieldValue := interfaceElem.Field(i)
+
+		value, err := util.SysReadFile(filepath.Join(devicePath, fieldType.Tag.Get("fileName")))
+
+		if err != nil {
+			if os.IsNotExist(err) || err.Error() == "operation not supported" || err.Error() == "invalid argument" {
```

Comment:
What's going on here with these other error conditions?

Reply:
was following the style in class_power_supply.go and net_class.go